- Spinlocks usage and "uninitialized values" fixed (compilation problems).
- Fixed STx710x minor configuration problem.
    
Signed-off-by: Pawel MOLL <pawel.moll@st.com>

diff --git a/sound/soc/stm/pcm_player.c b/sound/soc/stm/pcm_player.c
index 0fb5a3f..baf8e3b 100644
--- a/sound/soc/stm/pcm_player.c
+++ b/sound/soc/stm/pcm_player.c
@@ -401,6 +401,7 @@ static int snd_stm_pcm_player_prepare(struct snd_pcm_substream *substream)
 		bits_in_output_frame = 32;
 		break;
 	default:
+		bits_in_output_frame = 0; /* Avoid a -Os compilation warning */
 		snd_assert(0, return -EINVAL);
 		break;
 	}
diff --git a/sound/soc/stm/spdif_player.c b/sound/soc/stm/spdif_player.c
index e63c28d..e8e1508 100644
--- a/sound/soc/stm/spdif_player.c
+++ b/sound/soc/stm/spdif_player.c
@@ -692,6 +692,8 @@ static int snd_stm_spdif_player_copy(struct snd_pcm_substream *substream,
 				right_subframe <<= 8;
 				break;
 			default:
+				left_subframe = 0;  /* To avoid -Os */
+				right_subframe = 0; /* compilation warnings */
 				snd_assert(0, return -EINVAL);
 				break;
 			}
@@ -821,10 +823,10 @@ static int snd_stm_spdif_player_ctl_raw_get(struct snd_kcontrol *kcontrol,
 	snd_assert(spdif_player, return -EINVAL);
 	snd_stm_magic_assert(spdif_player, return -EINVAL);
 
-	spin_lock(spdif_player->modes_default_lock);
+	spin_lock(&spdif_player->modes_default_lock);
 	ucontrol->value.integer.value[0] = (spdif_player->input_mode_default
 			== SNDRV_STM_SPDIF_INPUT_MODE_RAW);
-	spin_unlock(spdif_player->modes_default_lock);
+	spin_unlock(&spdif_player->modes_default_lock);
 
 	return 0;
 }
@@ -847,10 +849,10 @@ static int snd_stm_spdif_player_ctl_raw_put(struct snd_kcontrol *kcontrol,
 	else
 		input_mode = SNDRV_STM_SPDIF_INPUT_MODE_NORMAL;
 
-	spin_lock(spdif_player->modes_default_lock);
+	spin_lock(&spdif_player->modes_default_lock);
 	changed = (input_mode != spdif_player->input_mode_default);
 	spdif_player->input_mode_default = input_mode;
-	spin_unlock(spdif_player->modes_default_lock);
+	spin_unlock(&spdif_player->modes_default_lock);
 
 	return changed;
 }
@@ -870,10 +872,10 @@ static int snd_stm_spdif_player_ctl_encoded_get(struct snd_kcontrol *kcontrol,
 	snd_assert(spdif_player, return -EINVAL);
 	snd_stm_magic_assert(spdif_player, return -EINVAL);
 
-	spin_lock(spdif_player->modes_default_lock);
+	spin_lock(&spdif_player->modes_default_lock);
 	ucontrol->value.integer.value[0] = (spdif_player->encoding_mode_default
 			== SNDRV_STM_SPDIF_ENCODING_MODE_ENCODED);
-	spin_unlock(spdif_player->modes_default_lock);
+	spin_unlock(&spdif_player->modes_default_lock);
 
 	return 0;
 }
@@ -896,10 +898,10 @@ static int snd_stm_spdif_player_ctl_encoded_put(struct snd_kcontrol *kcontrol,
 	else
 		encoding_mode = SNDRV_STM_SPDIF_ENCODING_MODE_PCM;
 
-	spin_lock(spdif_player->modes_default_lock);
+	spin_lock(&spdif_player->modes_default_lock);
 	changed = (encoding_mode != spdif_player->encoding_mode_default);
 	spdif_player->encoding_mode_default = encoding_mode;
-	spin_unlock(spdif_player->modes_default_lock);
+	spin_unlock(&spdif_player->modes_default_lock);
 
 	return changed;
 }
diff --git a/sound/soc/stm/stx710x.c b/sound/soc/stm/stx710x.c
index 22a7d4c..9b95769 100644
--- a/sound/soc/stm/stx710x.c
+++ b/sound/soc/stm/stx710x.c
@@ -237,7 +237,7 @@ static struct snd_stm_component __initdata snd_stm_stx710x_components[] = {
  */
 
 static union snd_stm_value __initdata number_0 = { .number = 0 };
-static union snd_stm_value __initdata number_1 = { .number = 0 };
+static union snd_stm_value __initdata number_1 = { .number = 1 };
 static union snd_stm_value __initdata list_2  = {
 	.list.len = 1,
 	.list.numbers = (int []) { 2 }
@@ -325,6 +325,7 @@ int __init snd_stm_stx710x_init(void)
 
 	default:
 		/* Unknown CPU! */
+		soc_type = NULL; /* To avoid a -Os compilation warning */
 		snd_assert(0, return -EINVAL);
 		break;
 	}
